firebase-installations/src/main/java/com/google/firebase/installations/local/FirebaseInstallationIdCache.java

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.installations.local;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.firebase.FirebaseApp;
+
+/**
+ * A layer that locally caches a few Firebase Installation attributes on top the Firebase
+ * Installation backend API.
+ */
+public class FirebaseInstallationIdCache {
+  // Status of each cache entry
+  // NOTE: never change the ordinal of the enum values because the enum values are stored in cache
+  // as their ordinal numbers.
+  public enum CacheStatus {
+    // Cache entry is synced to Firebase backend
+    REGISTERED,
+    // Cache entry is waiting for Firebase backend response or internal network retry
+    UNREGISTERED,
+    // Cache entry is in error state when syncing with Firebase backend
+    REGISTER_ERROR
+  }
+
+  private static final String SHARED_PREFS_NAME = "FirebaseInstallationIdCache";
+
+  private static final String FIREBASE_INSTALLATION_ID_KEY = "Fid";
+  private static final String CACHE_STATUS_KEY = "Status";
```

Comment:
Don't we want to cache to auth token, expiration time, and other stuff?

Reply:
Done. Thanks.